python-lib/dku_data_processing/format.py

```diff
@@ -0,0 +1,164 @@
+
+"""
+
+import logging
+import numpy as np
+import re
+
+from dku_data_processing.filtering import filter_dataframe
+from utils.df_column_naming import get_new_column_name
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s Custom Chart Geospatial Density  | %(levelname)s - %(message)s')
+
+
+def convert_numpy_int64_to_int(o):
+    if isinstance(o, np.int64):
+        return int(o)
+    raise TypeError
+
+
+def normalize(x):
+    """
+    Linear scaling of the values to set the percentile 12.5% to 0 and the percentile 87.5% to 1.
+    Doing such lower the impact of outliers on color scale.
+
+    :param x: Array of values to normalize
+    :return: the normalized array of values
+    """
+    left_percentile = np.percentile(x, 12.5)
+    right_percentile = np.percentile(x, 87.5)
+    if left_percentile == right_percentile:
+        return np.ones(len(x))
+    else:
+        x_scaled = (x - left_percentile) / (right_percentile - left_percentile)
+    return x_scaled
+
+
+def wkt_parser(wkt_point):
+    """
+    Parsing function of the Well Known Text point representation
+
+    :param wkt_point: <str>
+        example: `POINT(-73.97237 40.64749)`
+    :return: A couple of values (longitude, latitude)
+    """
+    if wkt_point is None:
+        return None, None
+    result = re.findall("POINT\(\s?(\S+)\s+(\S+)\s?\)", str(wkt_point))
+    if len(result) < 1:
+        return None, None
+    else:
+        return result[0]
+
+
+def prepare_df_to_display(df, detail, filters, geopoint, tooltips):
+    """
+    Extract minimum amount of columns and values to send to front-end.
+    Apply filters, compute normalization, parse latitude, longitude and build tooltip.
+
+    :param df: Input DataFrame
+    :param detail: Optional single column name of the column on which color will be computed
+    :param filters: Optional list of filters to apply to the DataFrame to select values
+    :param geopoint: Name of the column containing GeoPoint WKT objects
+    :param tooltips: List of columns to include in the tooltip value that will be displayed
+    :return: Data to send to front-end
+    """
+    # Get filtered dataframe
+    if filters:
+        df = filter_dataframe(df, filters)
+    if df.empty:
+        # Handle case where dataframe is empty after filtering
+        return []
+
+    new_detail_column = get_new_column_name(df, "detail")
+
+    # Get normalized detail column
+    if (detail is None) or (df.dtypes[detail] not in ['int64', 'float64']):
+        df[new_detail_column] = 1
+    else:
+        df[new_detail_column] = normalize(df[detail].values)
+    if (detail is not None) and (df.dtypes[detail] not in ['int64', 'float64']):
+        raise ValueError("The detail column should be a int or float type.")
+
+    def convert(x):
+        if x is not None:
+            return float(x)
+        else:
+            return None
```

Comment:
Simplification proposal: 
```suggestion
    convert = lambda x: x and float(x)
```

Reply:
Done in 4cf4276